[RAPPS-DB] Add Vi IMproved 8.2.5171

nospam3021 opened the pull request:
Using last 8.2 build, as newer 9.0 and 9.1 versions fail to launch due to KERNEL32.dll missing GetFinalPathNameByHandleW (not fixed by [CORE-14576][]).

[CORE-14576]: https://jira.reactos.org/browse/CORE-14576

binarymaster commented on vim.txt:
```diff
@@ -0,0 +1,10 @@
+[Section]
+Name = Vi IMproved
```
```suggestion
Name = Vim (Vi IMproved)
```

binarymaster commented on vim.txt:
```diff
@@ -0,0 +1,10 @@
+[Section]
+Name = Vi IMproved
+Version = 8.2.5171
+License = GPL-compatible charityware
```
```suggestion
License = GPL-compatible charityware
LicenseType = 1
```

binarymaster commented on vim.txt:
```diff
@@ -0,0 +1,10 @@
+Version = 8.2.5171
+License = GPL-compatible charityware
+Description = Vim is a highly configurable text editor built to enable efficient text editing. It is an improved version of the vi editor distributed with most UNIX systems.
+Category = 7
```
Not sure about category 7 (Development), since for example Notepad++ uses 6 (Office).

learn-more replied:
I'd argue that Notepad++ is in the wrong category then.

binarymaster reviewed (commented):
Suggestions for improvements: add icon, screenshot, and maybe AMD64 section if that works in ReactOS x64.

nospam3021 commented:
Thanks @binarymaster for reviewing!

I've updated the branch with a _fixup!_ commit addressing most of your comments. Dunno how squashing branches usually are done within the ReactOS project. Should I force-push the branch once there are no more comments, is there some pipeline setup to take care of it automatically, or what's the typical procedure?

Finding the most suitable category indeed is tricky. Looking at comparable software, we see: AkelPad @ 6, EverEdit @ 7, Nano for Windows @ 12, Notepad2e @ 7, Notepad++ @ 6, Programmer's Notepad @ 7, PSPad @ 6, Sublime Text 3 @ 7. Moving to category 6 sure seems reasonable.

The `vim.ico` file comes from [dbf8094](https://github.com/vim/vim/tree/dbf8094/src), as the (higher resolution) file later introduced in v8.2.4545 appears to fail rendering correctly with ReactOS.

Screenshots could be neat, but can't really see where to host such pictures.

I hope my understanding of how to do the architecture specific changes is correct. Unfortunately I only have access to a 32-bit x86 machine, so that part is not actually tested.

binarymaster commented on vim.txt:
```diff
@@ -0,0 +1,16 @@
+URLDownload = https://github.com/vim/vim-win32-installer/releases/download/v8.2.5171/gvim_8.2.5171_x86_signed.exe
+SHA1 = 40b579d4ea767a533524256a289e4e14eb8a4c81
+SizeBytes = 9841432
+
```
```suggestion
Icon = vim.ico

```

nospam3021 replied:
The text under [Icon](https://reactos.org/wiki/RAPPS#Icon) in the wiki reads:

> if the filename of the .txt file matches the filename of an .ico file, that icon will also be displayed.

Suggesting that the addition of the field would technically be a pointless declaration. I did try placing the file in `appdb/` and it did show up before pushing the change. Furthermore, out of the currently existing `.ico` files of this repository; 11 do not share their basename with any `.txt` file; of those who do, 41 rely on the filename match rule and 23 explicitly add the _Icon_ property. I did not analyze the history. Is there a preference in moving towards specifying the icon file purely for documentation purposes?

learn-more replied:
There is no preference currently, so omitting it is perfectly fine

binarymaster replied:
Didn't knew that, it's alright then. 🙂

binarymaster commented:
> Dunno how squashing branches usually are done within the ReactOS project.

PRs at rapps-db are usually squashed & merged by maintainers from GitHub Web UI, so no need to worry about it.

> Screenshots could be neat, but can't really see where to host such pictures.

Looking at https://github.com/search?q=repo%3Areactos%2Frapps-db%20Screenshot&type=code

It's usually either imgur, some GitHub source repos, or GitHub CDN (attach screens to PR, then use their image links).

> Unfortunately I only have access to a 32-bit x86 machine, so that part is not actually tested.

Maybe @julenuri could test?

julenuri commented:
![gvim-2](https://github.com/reactos/rapps-db/assets/27921286/fa4d9e3f-11c4-4c00-aa6e-24be0c4ff0c3)
![gvim-1](https://github.com/reactos/rapps-db/assets/27921286/af4ce1b7-e9d3-4d9f-838e-c030e56e824d)

binarymaster commented:
> However I don't find it so much... Informative, it's VIM. Too tiny.

Is x64 version working fine?

nospam3021 commented:
@julenuri I agree, not informative enough to be motivated. Yet for completeness; if you were keen on making another set of screenshots, running `:set guifont=*` is a quick way to make them appear less tiny. One might perhaps also wish do something like `:colorscheme darkblue`. Yet I'm not sure its too important. The images included above give me 404 when I try opening them outside of this thread (oh the wonders of contemporary web).

nospam3021 commented:
For what it's worth, I made a couple of failed attempts at installing ReactOS on a x64 virtual machine. However the iso for 7550 (the same version I used for x86) does not even boot, and version 7555 fails during installation. I am afraid the x64 rabbit hole seems a bit deeper than I'm prepared to go down in.

binarymaster commented on vim.txt:
```diff
@@ -0,0 +1,16 @@
+SizeBytes = 9841432
+
+[Section.amd64]
+URLDownload = https://github.com/vim/vim-win32-installer/releases/download/v8.2.5171/gvim_8.2.5171_x64_signed.exe
+SHA1 = bfb2f956d91663b647c403801d031af1fca95915
+SizeBytes = 10411736
```
Installer is not 64-bit native application:

![VirtualBox_ReactOS_07_02_2024_20_40_12](https://github.com/reactos/rapps-db/assets/578406/b123d3f7-5e66-4337-bfd3-99116939cad1)


```suggestion
URLDownload = https://github.com/vim/vim-win32-installer/releases/download/v8.2.5171/gvim_8.2.5171_x64_signed.zip
SHA1 = e74858e328ee1d1d05c9dd66dfb366b89ada89e1
SizeBytes = 20577877
```

binarymaster commented on vim.txt:
```diff
@@ -0,0 +1,16 @@
+[Section]
+Name = Vim (Vi IMproved)
```
Improvement based on the program output:
```suggestion
Name = VIM - Vi IMproved
```

binarymaster commented:
Made better screenshots:

![](https://i.imgur.com/PzitUVC.png)

![](https://i.imgur.com/IWTOJH1.png)

binarymaster commented on vim.txt:
```diff
@@ -0,0 +1,16 @@
+Description = Vim is a highly configurable text editor built to enable efficient text editing. It is an improved version of the vi editor distributed with most UNIX systems.
+Category = 7
+URLSite = http://www.vim.org/
+URLDownload = https://github.com/vim/vim-win32-installer/releases/download/v8.2.5171/gvim_8.2.5171_x86_signed.exe
```
```suggestion
URLDownload = https://github.com/vim/vim-win32-installer/releases/download/v8.2.5171/gvim_8.2.5171_x86_signed.exe
Screenshot1 = https://i.imgur.com/PzitUVC.png
```

binarymaster commented on vim.txt:
```diff
@@ -0,0 +1,16 @@
+SizeBytes = 9841432
+
+[Section.amd64]
+URLDownload = https://github.com/vim/vim-win32-installer/releases/download/v8.2.5171/gvim_8.2.5171_x64_signed.zip
```
```suggestion
URLDownload = https://github.com/vim/vim-win32-installer/releases/download/v8.2.5171/gvim_8.2.5171_x64_signed.zip
Screenshot1 = https://i.imgur.com/IWTOJH1.png
```
